### ISSUE TITLE
PyFFTW DCTs

### ISSUE THREAD

**Issue by jo1763** [pyFFTW/pyFFTW#137]
Good Morning, 

I'm new to PyFFTW and I've tried to google this question without success. Is there a way to use PyFFTW to import any of FFTWs DCTs? Perhaps there is an obvious implementation but I haven't found it. 

Thank you 

Jason

**Comment by drwells** [comment]
Yes (see #95). I need to finish this at some point but I have not had time to do so (the only thing left is writing some tests).

**Comment by crowsonkb** [comment]
I was just looking for these yesterday... FFTW's DCTs are the only ones I've found that are reasonably fast on prime-sized inputs and I need to avoid denial of service attacks in a web app. I'd really love to be able to use them from Python.

**Comment by d-mueller** [comment]
Bump. I need DCTs too and I'm currently stuck with the ones from scipy. Would be really great to have them in pyFFTW!

**Comment by rileyjmurray** [comment]
I'm also interested in DCT support.

In the past few years, randomized methods for numerical linear algebra have been shown to be outperform classical methods on certain problems (e.g. least-squares ``min || A @ x - b  ||`` where ``A`` is very tall). The trouble is that the randomized methods require fast DCT to outperform the classical methods by wallclock time.

**Comment by hgomersall** [comment]
@rileyjmurray I think we're in the situation that #95 could do with some assistance so it can be merged. I daresay you could contribute, or fork and resubmit.

**Comment by hgomersall** [comment]
Actually #256 is the current point of development (built on #95).

**Comment by rileyjmurray** [comment]
In principle I'm happy to help, but I'm swamped for the next month at least. Also, I have little experience writing C, and no experience writing C <--> python interface code. From the looks of the PR maybe there isn't much C code that remains to be written. I just made a comment there asking about TODOs.
